index.d.ts

```diff
@@ -159,7 +159,17 @@ declare namespace meow {
 		readonly hardRejection?: boolean;
 	}
 
-	interface Result {
+	type TypedFlags<Flags extends MinimistOptions> = {
+		[F in keyof Flags]: Flags[F] extends { type: 'number' } | 'number'
```

Comment:
Why the ` | 'number'` part? `flags` has to be an object.

Reply:
When I created this you could also define flags like `flags: { foo: 'number' }`. So here I'm checking if a flag is the string `'number'` or an object with `type: 'number'`. Did that change?

Reply:
Yeah. That's not supported anymore.

Reply:
OK. I've updated the code to reflect that, but `MinimistOptions` still allows defining `{flags: {foo: 'number'}}`. The possibility comes from

```ts
type Options<ArrayOptionContentType = unknown> = {
  [key: string]: OptionType /* | <other object types> */
type OptionType = 'string' | 'boolean' | 'number' | 'array';
```

Passing in `foo: 'number'` still works because `meow` is just calling `yargs`. Should the types be stricter to only allow the documented CLI? Reusing `MinimistOptions` is easy (and correct with the current codebase), but it doesn't align with the docs.

In any case, I don't think this is blocking this PR so we could move the discussion to a new issue, if it's even worth continuing the discussion.

Reply:
> Should the types be stricter to only allow the documented CLI?

Yes, ideally we should that kind of usage in the JS code too.

Reply:
> ideally we should that kind of usage in the JS code too.

I'm guessing you meant forbid here since the JS code already allows patterns that are not part of the docs. If I have time maybe I'll make a PR to first restrict the types.

Reply:
Yes, forbid.